### DISCUSSION TITLE
Pin Assignment for any ESP32 and LAN8720

### DISCUSSION THREAD

**Discussion by martinmmi** [khoih-prog/WebServer_WT32_ETH01#18] [Q&A]
Hello, i am Martin from Germany and i have one basic question. Since few weeks I try to connect an **LAN8720** to an **LILYOGO ESP32**. Its not so easy because the LAN8720 Board needs much external signals from the esp.

**For Example:**
- MDC / MDIO (I²C Bus)
- CRS
- RX1
- RX0
- nINT (CLOCK)
etc.

**Now my question:** Where can i do the pin assignments in the [WebServer_WT32_ETH01] Lib? Because every pinout is different on microcontrollers. Maybe for the example: AdvancedWebServer.ino

I have that controller concrete:
http://www.lilygo.cn/prod_view.aspx?TypeId=50003&Id=1271&FId=t3:50003:3

Thank you so much for reply in advance.  I'll show my appreciation if this works.

**Comment by khoih-prog** [comment]
HI @martinmmi 

Thanks for your interest in the library.

> Its not so easy because the LAN8720 Board needs much external signals from the esp.

I suggest you use the better and simpler SPI-based **W5500 Ethernet** with either [**Ethernet_Generic**](https://github.com/khoih-prog/Ethernet_Generic) or Async-ready [**WebServer_ESP32_W5500**](https://github.com/khoih-prog/WebServer_ESP32_W5500)

> Where can i do the pin assignments in the [WebServer_WT32_ETH01] Lib?

Sorry I can't help here as the issue is not popular. You can research by yourself or ask question on some Forum if to use LAN8720 is a **must**.

Good Luck,
